Testing.Rmd

```diff
@@ -207,65 +286,93 @@ A test failure looks something like this:
     `expected`: "0.0.0.9000"
 
 Each failure gives a description of the test (e.g., "get_release_data() works if no file found"), its location (e.g., "test-release.R:108:3"), and the reason for the failure (e.g., "res$Version (`actual`) not equal to "0.0.0.9000" (`expected`)").
-The goal is to pass all the tests.
+
+The idea is that you'll modify your code (either the functions defined below `R/` or the tests in `tests/testthat/`) until all tests are passing.
 
 ## Test structure {#test-structure}
 
 A test file lives in `tests/testthat/`.
 Its name must start with `test`.
-Here's an example of a test file from the stringr package:
+We will inspect and execute a test file from the stringr package.
+
+But first, for the purposes of rendering this book, we must attach stringr and testthat.
```

Comment:
Should we add a block that is "we're running this code so the book work, but you should never do this"?

Reply:
https://github.com/hadley/r-pkgs/issues/778